packages/react/src/components/dateInput/components/datePicker/types/DayPickerProps.ts

```diff
@@ -11,6 +11,16 @@ export type DayClickEventHandler = (day: Date, e: React.MouseEvent) => void;
  */
 export type MonthChangeEventHandler = (month: Date, e: React.MouseEvent) => void;
 
+/**
+ * Props for legend item.
+ * @param {{ id: string, color: string, label: string }}
+ */
+export type LegendItem = {
+  id: string;
+  color: string;
```

Comment:
Hmm, you have a good point. Commonly only `id` prop is used in HDS components but maybe we can make an exception here because it might not be clear what the id means in this context. The prefix shouldn't be added here because user needs the id when setting the `setDateAriaDescribedBy` function for specific dates.